legend-sdlc-generation-service/src/main/java/org/finos/legend/sdlc/generation/service/ServiceExecutionClassGenerator.java

```diff
@@ -253,6 +267,70 @@ private void addImport(Class<?> cls)
+                "        }\n" +
+                (parameters.size() > 0 ?
+                        IntStream.range(0, parameters.size())
+                                .mapToObj(i ->
+                                {
+                                    ExecutionParameter p = parameters.get(i);
+                                    String type = p.getTypeString();
+                                    String valueExtractor =
+                                            "long".equals(type) || "Long".equals(type) ?
+                                                    "(args.get(" + i + ") instanceof Number ? ((Number) args.get(" + i + ")).longValue() : args.get(" + i + "))" :
+                                                    "double".equals(type) || "Double".equals(type) ?
+                                                            "(args.get(" + i + ") instanceof Number ? ((Number) args.get(" + i + ")).doubleValue() : args.get(" + i + "))" :
+                                                            "args.get(" + i + ")";
+
+
+                                    return "        " + type + " param" + i + ";\n" +
+                                            "        try\n" +
+                                            "        {\n" +
+                                            "            param" + i + " = args.get(" + i + ") == null ? null : (" + type + ") " + valueExtractor + ";\n" +
+                                            "        }\n" +
+                                            "        catch (ClassCastException e)\n" +
+                                            "        {\n" +
+                                            "            throw new IllegalArgumentException(\"Unexpected parameter type. '" + p.legendParamName + "' parameter (index: " + i + ") should be of type '" + type + "'\");\n" +
+                                            "        }\n" +
+                                            "        catch (NullPointerException e)\n" +
+                                            "        {\n" +
+                                            "            throw new IllegalArgumentException(\"Unexpected parameter value. '" + p.legendParamName + "' parameter (index: " + i + ") should not be null\");\n" +
+                                            "        }\n";
+                                })
+                                .collect(Collectors.joining("\n", "\n", "\n")) :
+                        "\n"
+                ) +
+                "        newExecutionBuilder(" + parameters.size() + ")\n" +
+                IntStream.range(0, parameters.size()).mapToObj(i -> "            .withParameter(\"" + parameters.get(i).legendParamName + "\", param" + i + ")").collect(Collectors.joining("\n", "", parameters.size() > 0 ? "\n" : "")) +
+                "            .withServiceRunnerInput(serviceRunnerInput)\n" +
+                "            .executeToStream(outputStream);\n" +
```

Comment:
I don't think there should be any calls here that create new threads. Honestly, I think AbstractServicePlanExecutor should not have any methods that create new threads. Questions of threading should be left up to the caller, not the service execution classes.

Reply:
The plan is to have the 'run' method with output stream execute in an async fashion by default and hence the thread creation. I'll discuss regarding this with @pierredebelen to see if it makes sense and get it changed in AbstractServicePlanExecutor.

Reply:
It's much better to let the user call it asynchronously if they want that. That way the calling application manages the threads, which is more appropriate. And it's not like it's difficult for callers to call it asynchronously. Indeed, it is quite easy.

On top of that, the current approach gives no feedback on when the execution finishes or whether it finishes successfully. And if an error is thrown during the execution, it is lost to the caller. If we leave it to the caller whether to execute asynchronously, they can use an ExecutorService which will solve all those problems, as well as allowing them to manage the threads.

Reply:
Changing it here https://github.com/finos/legend-engine/pull/356